internal/runner/runner.go

```diff
@@ -140,9 +143,21 @@ func (r *CommandRunner) runStep(ctx context.Context, step *CommandStep, dry bool
 	}
 
 	return &CommandStep{
-		Command: append([]string{name}, args...),
-		// NOTE: We don't actually include POSIX env vars as they are redundant.
-		Env:        env,
+		Command:    append([]string{name}, args...),
+		Env:        userEnv,
 		WorkingDir: pwd,
 	}, nil
 }
+
+func dedupEnv(env []string) []string {
+	var deduped []string
+	seen := map[string]bool{}
+	for i := len(env) - 1; i >= 0; i-- {
+		k, _, found := strings.Cut(env[i], "=")
+		if found && !seen[k] {
```

Comment:
The `exec` package [leaves in malformed env var strings](https://cs.opensource.google/go/go/+/refs/tags/go1.19.4:src/os/exec/exec.go;l=1247-1252;drc=888047c31007c0ad96764254fa8afb1417ddfd40). Should we do that too? or produce some kind of error?

Reply:
Yeah, I think going with what they did is safest.